### PR TITLE
Standardize Output Variable and Meter Fuel Types

### LINKED ISSUE #7622: Standardize Output Variable and Meter Fuel Types
**This is part 4 of 4: Part 1 = #7532, Part 2 = #7523 , and Part 3 is #7621.**

Issue overview
--------------
This is the final piece of standardizing fuel type on both input and output. Issues #5941 and #7621 dealt with the input side, with the goal of no diffs. This issue takes the plunge and modifies output variable names and meter names to the final standard list:

**Input choice, Meter Resource and Output Variable, Table Headings** (*updated 03 Aug 2020 MJW*)
Electricity, Electricity, Electricity
NaturalGas, NaturalGas, Natural Gas
Propane, Propane, Propane
Diesel, Diesel, Diesel
Gasoline, Gasoline, Gasoline
FuelOilNo1, FuelOilNo1, Fuel Oil No 1
FuelOilNo2, FuelOilNo2, Fuel Oil No 2
Coal, Coal, Coal
OtherFuel1, OtherFuel1, Other Fuel 1
OtherFuel2, OtherFuel2, Other Fuel 2
Water, Water, Water

Note that Meter Resource = Input Choice when this is all done. That may allow some simplification in how the objects use fuel types.

This will impact:

1. Meter resource types for Gas will become NaturalGas.

2. Output variable names will change for Electricity, NaturalGas, FuelOilNo1, FuelOilNo2, OtherFuel1, and OtherFuel2 ( more less?). This will required potentially many lines in the transition Report Variables 9-2-0 to 9-3-0.csv - script this? Or see modify transition to replace on substrings - maybe that's better.

3. Electricity output variables are currently different from other fuels. "Electric Power" and "Electric Energy" vs others that are "Gas Rate" and "Gas Energy". So, if we go with always matching the long version of the input fuel type, then these would become "Electricity Rate" and "Electricity Energy" - a little less natural, but makes things consistent across all fuel types. ~Yes? No? Some other approach?~ *updated 03 Aug 2020 MJW - Yes, change to Electricity Rate and Electricity Energy.*

4. For the most part table output column headings are using the desired strings, but some column headings may be affected and some table row headings may change.

5. There will be diffs everywhere:  mtd, eso, mtr, table - even though it's more work, it may be best to stage this in 2 parts: change the meters first (there should be no eso diffs) then change the output variables (there should be no mtr diffs). Or make a special version of mathdiff that can map old to new variable names?

6. More???

### Details
Some additional details for this issue (if relevant):
 - Followup to #7621 (final part of #5941)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by dareumnam**
Pull request overview
---------------------

 - Fixes #7622
   - Output variable names and meters were changed, mostly `"Electric"` to `"Electricity"` and `"Gas"` to `"NaturalGas"`.
Input choice, Meter Resource and Output Variable, Table Headings
Electricity, Electricity, Electricity
NaturalGas, NaturalGas, Natural Gas
Propane, Propane, Propane
Diesel, Diesel, Diesel
Gasoline, Gasoline, Gasoline
FuelOilNo1, FuelOilNo1, Fuel Oil No 1
FuelOilNo2, FuelOilNo2, Fuel Oil No 2
Coal, Coal, Coal
OtherFuel1, OtherFuel1, Other Fuel 1
OtherFuel2, OtherFuel2, Other Fuel 2
Water, Water, Water
   - Changed the expanded names (Natural Gas) to the condensed names(NaturalGas) for output variables (ex: `"Boiler NaturalGas Energy"`)
   - Changed electricity output variables to `"Electricity Rate"` and `"Electricity Energy"` from `"Electric Power"` and `"Electric Energy"`, to make things consistent across all fuel types.
   - Renamed Output:Variables and Output:Meter report variables in the test files impacted by the name changes.
   - Updated one existing unit test case in `OutputProcessor.unit.cc`
 - Separated three output variables that were summing NaturalGas and Propane in SystemReports.cc
   - Three output variables were changed and three output variables were created:
`"Air System Gas Energy"` → `"Air System NaturalGas Energy"` and `"Air System Propane Energy"` were created
`"Air System Heating Coil Gas Energy"` → `"Air System Heating Coil NaturalGas Energy"` and `"Air System Heating Coil Propane Energy"` were created
`"Air System Humidifier Gas Energy"` → `"Air System Humidifier NaturalGas Energy"` and `"Air System Humidifier Propane Energy"` were created 
   - Created `SystemReports.unit.cc`, added a unit test, and updated `CMakeLists.txt` 
   - Updated `Report Variables 9-3-0 to 9-4-0.csv`
 - Diffs
   - ERR diffs were from renamed output variables
   - AUD diffs & RDD diffs from the three output variables that separated
`"Air System Gas Energy"` → `"Air System NaturalGas Energy"` and `"Air System Propane Energy"`
`"Air System Heating Coil Gas Energy"` → `"Air System Heating Coil NaturalGas Energy"` and `"Air System Heating Coil Propane Energy"`
`"Air System Humidifier Gas Energy"` → `"Air System Humidifier NaturalGas Energy"` and `"Air System Humidifier Propane Energy"`
   - 20 regression testfiles were randomly selected and locally tested. Confirmed that NO csv output differences except column head names.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [X] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [X] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [X] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DXCoils.cc [hunk 1]
```diff
@@ -187,7 +187,7 @@ namespace DXCoils {
     int const NumValidOutputFuelTypes(9);
     Array1D_string const
         cValidOutputFuelTypes(NumValidOutputFuelTypes,
-                              {"Electricity", "Gas", "Propane", "Diesel", "Gasoline", "FuelOilNo1", "FuelOilNo2", "OtherFuel1", "OtherFuel2"});
+                              {"Electricity", "NaturalGas", "Propane", "Diesel", "Gasoline", "FuelOilNo1", "FuelOilNo2", "OtherFuel1", "OtherFuel2"});
```
Comment (mjwitte) [inline src/EnergyPlus/DXCoils.cc:190]: Let's not change this now, but we should be able to remove this and use some form of `UtilityRoutines::validateFuelType...` instead.

File: src/EnergyPlus/ExteriorEnergyUse.cc [hunk 2]
```diff
@@ -476,10 +476,10 @@ namespace ExteriorEnergyUse {
         // Select the correct Number for the associated ascii name for the fuel type
         if (UtilityRoutines::SameString(FuelTypeAlpha, "Electricity")) {
             FuelTypeNumber = ExteriorFuelUsage::ElecUse;
-            FuelTypeString = "Electric";
+            FuelTypeString = "Electricity";
```
Comment (mjwitte) [inline src/EnergyPlus/ExteriorEnergyUse.cc:479]: Another place that should be change over to use the common function - but not in this PR.

File: src/EnergyPlus/FuelCellElectricGenerator.cc [hunk 3]
```diff
@@ -1061,9 +1061,9 @@ namespace FuelCellElectricGenerator {
 
     void FCDataStruct::setupOutputVars()
     {
-        SetupOutputVariable("Generator Produced Electric Power", OutputProcessor::Unit::W, this->Report.ACPowerGen, "System", "Average", this->Name);
+        SetupOutputVariable("Generator Produced Electricity Rate", OutputProcessor::Unit::W, this->Report.ACPowerGen, "System", "Average", this->Name);
 
-        SetupOutputVariable("Generator Produced Electric Energy",
+        SetupOutputVariable("Generator Produced Electricity Energy",
```
Comment (mjwitte) [inline src/EnergyPlus/FuelCellElectricGenerator.cc]: I don't see these in the Report Variables csv file. These should probably convert to "Generator Produced AC Electricity Rate/Energy" for consistency with the DC power below.

File: src/EnergyPlus/HVACVariableRefrigerantFlow.cc [hunk 4]
```diff
@@ -159,7 +159,7 @@ namespace HVACVariableRefrigerantFlow {
     int const NumValidFuelTypes(9);
     Array1D_string const
         cValidFuelTypes(NumValidFuelTypes,
-                        {"Electric", "NaturalGas", "Propane", "Diesel", "Gasoline", "FuelOilNo1", "FuelOilNo2", "OtherFuel1", "OtherFuel2"});
+                        {"Electricity", "NaturalGas", "Propane", "Diesel", "Gasoline", "FuelOilNo1", "FuelOilNo2", "OtherFuel1", "OtherFuel2"});
```
Comment (mjwitte) [inline src/EnergyPlus/HVACVariableRefrigerantFlow.cc:162]: Another place to remove - later.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc [hunk 5]
```diff
@@ -1916,43 +1916,43 @@ void InputProcessor::addVariablesForMonthlyReport(std::string const &reportName)
         addRecordToOutputVariableStructure("*", "ZONE THERMAL COMFORT FANGER MODEL PPD");
 
     } else if (reportName == "CHILLERREPORTMONTHLY") {
-        addRecordToOutputVariableStructure("*", "CHILLER ELECTRIC ENERGY"); // on meter
-        addRecordToOutputVariableStructure("*", "CHILLER ELECTRIC POWER");
+        addRecordToOutputVariableStructure("*", "CHILLER ELECTRICITY ENERGY"); // on meter
+        addRecordToOutputVariableStructure("*", "CHILLER Electricity Rate");
```
Comment (mjwitte) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc]: This probably needs to be UPPERCASE.

File: src/EnergyPlus/InputProcessing/InputProcessor.cc [hunk 6]
```diff
@@ -1916,43 +1916,43 @@ void InputProcessor::addVariablesForMonthlyReport(std::string const &reportName)
...
+        addRecordToOutputVariableStructure("*", "BOILER NATURALGAS CONSUMPTION"); // on meter
         addRecordToOutputVariableStructure("*", "BOILER HEATING ENERGY");  // on meter
         addRecordToOutputVariableStructure("*", "BOILER HEATING RATE");
-        addRecordToOutputVariableStructure("*", "BOILER GAS CONSUMPTION RATE");
+        addRecordToOutputVariableStructure("*", "BOILER NATURALGAS CONSUMPTION RATE");
         addRecordToOutputVariableStructure("*", "BOILER INLET TEMPERATURE");
         addRecordToOutputVariableStructure("*", "BOILER OUTLET TEMPERATURE");
         addRecordToOutputVariableStructure("*", "BOILER MASS FLOW RATE");
-        addRecordToOutputVariableStructure("*", "BOILER ANCILLARY ELECTRIC POWER");
+        addRecordToOutputVariableStructure("*", "BOILER ANCILLARY Electricity Rate");
```
Comment (mjwitte) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc]: Uppercase

File: src/EnergyPlus/InputProcessing/InputProcessor.cc [hunk 7]
```diff
@@ -1916,43 +1916,43 @@ void InputProcessor::addVariablesForMonthlyReport(std::string const &reportName)
...
-        addRecordToOutputVariableStructure("*", "COOLING COIL CRANKCASE HEATER ELECTRIC ENERGY");
+        addRecordToOutputVariableStructure("*", "COOLING COIL CRANKCASE HEATER ELECTRICITY ENERGY");
         addRecordToOutputVariableStructure("*", "COOLING COIL RUNTIME FRACTION");
         addRecordToOutputVariableStructure("*", "COOLING COIL TOTAL COOLING RATE");
         addRecordToOutputVariableStructure("*", "COOLING COIL SENSIBLE COOLING RATE");
         addRecordToOutputVariableStructure("*", "COOLING COIL LATENT COOLING RATE");
-        addRecordToOutputVariableStructure("*", "COOLING COIL ELECTRIC POWER");
-        addRecordToOutputVariableStructure("*", "COOLING COIL CRANKCASE HEATER ELECTRIC POWER");
+        addRecordToOutputVariableStructure("*", "COOLING COIL Electricity Rate");
+        addRecordToOutputVariableStructure("*", "COOLING COIL CRANKCASE HEATER Electricity Rate");
```
Comment (mjwitte) [inline src/EnergyPlus/InputProcessing/InputProcessor.cc]: more uppercase . . . look for more below.

File: src/EnergyPlus/OutputReportPredefined.cc [hunk 8]
```diff
@@ -1931,13 +1931,13 @@ namespace OutputReportPredefined {
         pdstDXCoolCoil2 = newPreDefSubTable(pdrEquip, "DX Cooling Coil ASHRAE 127 Standard Ratings Report");
         pdchDXCoolCoilType = newPreDefColumn(pdstDXCoolCoil2, "DX Cooling Coil Type");
         pdchDXCoolCoilNetCapSIA = newPreDefColumn(pdstDXCoolCoil2, "Rated Net Cooling Capacity Test A [W]");
-        pdchDXCoolCoilElecPowerA = newPreDefColumn(pdstDXCoolCoil2, "Rated Electric Power Test A [W]");
+        pdchDXCoolCoilElecPowerA = newPreDefColumn(pdstDXCoolCoil2, "Rated Electricity Rate Test A [W]");
```
Comment (mjwitte) [inline src/EnergyPlus/OutputReportPredefined.cc]: Hmmm, for these ratings tables I wonder if we should leave these as-is. @rraustad @Myoldmopar @Nigusse Do these heading match terminology in the rating standard?

File: src/EnergyPlus/Pumps.cc [hunk 9]
```diff
@@ -1084,19 +1084,19 @@ namespace Pumps {
             if (PumpEquip(PumpNum).PumpType == Pump_VarSpeed || PumpEquip(PumpNum).PumpType == Pump_ConSpeed ||
                 PumpEquip(PumpNum).PumpType == Pump_Cond) {
 
-                SetupOutputVariable("Pump Electric Energy",
+                SetupOutputVariable("Pump Electricity Energy",
```
Comment (mjwitte) [inline src/EnergyPlus/Pumps.cc:1087]: I don't see Pump Electricity in the report variables csv.

File: src/EnergyPlus/PollutionModule.cc [hunk 10]
```diff
@@ -758,87 +758,87 @@ namespace PollutionModule {
                     Pollution.NatGasCoef.Source = rNumericArgs(2);
                     if (!lAlphaFieldBlanks(3)) {
                         CheckFFSchedule(
-                            cCurrentModuleObject, "Natural Gas", cAlphaFieldNames(3), cAlphaArgs(3), Pollution.NatGasCoef.SourceSched, ErrorsFound);
+                            cCurrentModuleObject, "NaturalGas", cAlphaFieldNames(3), cAlphaArgs(3), Pollution.NatGasCoef.SourceSched, ErrorsFound);
```
Comment (mjwitte) [inline src/EnergyPlus/PollutionModule.cc:761]: For later - in GetPollutionFactorInput - line 1116 - remove synonyms from if statement.

File: src/EnergyPlus/OutputProcessor.cc [hunk 11]
```diff
@@ -1920,7 +1920,7 @@ namespace OutputProcessor {
                 OutResourceType = "Electricity";
 
             } else if (meterType == "GAS" || meterType == "NATURALGAS") {
-                OutResourceType = "Gas";
+                OutResourceType = "NaturalGas";
```
Comment (mjwitte) [inline src/EnergyPlus/OutputProcessor.cc:1923]: For later - remove synonyms from the if statements in this function (gas, electric, propanegas, etc.)

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 12]
```diff
@@ -2329,10 +2329,10 @@ namespace WaterThermalTanks {
             {
                 auto const SELECT_CASE_var(DataIPShortCuts::cAlphaArgs(4));
                 if (SELECT_CASE_var == "ELECTRICITY") {
-                    Tank.FuelType = "Electric";
+                    Tank.FuelType = "Electricity";
```
Comment (mjwitte) [inline src/EnergyPlus/WaterThermalTanks.cc:2332]: Another possible place to use the common function? (later)

File: src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv [hunk 13]
```diff
@@ -1,114 +1,490 @@
...
-Environmental Impact Fuel Oil #1 PM2.5 Emissions Mass,Environmental Impact FuelOilNo1 PM2.5 Emissions Mass,
-Environmental Impact Fuel Oil #1 NH3 Emissions Mass,Environmental Impact FuelOilNo1 NH3 Emissions Mass,
-Environmental Impact Fuel Oil #1 NMVOC Emissions Mass,Environmental Impact FuelOilNo1 NMVOC Emissions Mass,
-Environmental Impact Fuel Oil #1 Hg Emissions Mass,Environmental Impact FuelOilNo1 Hg Emissions Mass,
-Environmental Impact Fuel Oil #1 Pb Emissions Mass,Environmental Impact FuelOilNo1 Pb Emissions Mass,
-Environmental Impact Fuel Oil #1 Water Consumption Volume,Environmental Impact FuelOilNo1 Water Consumption Volume,
-Environmental Impact Fuel Oil #1 Nuclear High Level Waste Mass,Environmental Impact FuelOilNo1 Nuclear High Level Waste Mass,
-Environmental Impact Fuel Oil #1 Nuclear Low Level Waste Volume,Environmental Impact FuelOilNo1 Nuclear Low Level Waste Volume,
-old variable name,new variable name -- <DELETE> to delete,add variable names (before this line) and leave off units -- text in this column will add a warning
+9.3.0,9.4.0,Transition notes - some of these are EMS variable names,,
```
Comment (mjwitte) [inline src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv]: See notes highlighted in yellow in the attached copy of this file. 
[Report Variables 9-3-0 to 9-4-0 - MJW Notes.xlsx](https://github.com/NREL/EnergyPlus/files/5099654/Report.Variables.9-3-0.to.9-4-0.-.MJW.Notes.xlsx)

**Review by mjwitte — COMMENTED** [review]
@dareumnam Wow! A huuuuge effort here - thanks for getting this pulled together. It's very clean for the most part. I've made a few comments here and there. Also there's some crossover with the fuel oil work in the report variables files. See notes in the report variables attachment below.

@Myoldmopar I haven't done any transition testing on this. At some point (probably after I/O Freeze) I'd like to see a full test suite transition (just the report variables stuff) with a modified mathdiff/tablediff that ignores differences in the column headings (maybe that could be a future option) just to gain confidence that there are no csv diffs.

Oh yes, and a *minor* detail - the I/O Ref and possibly other docs will need changes. But that can happen after I/O freeze.

File: src/EnergyPlus/DXCoils.cc (hunk 1 above)
Comment (dareumnam) [reply src/EnergyPlus/DXCoils.cc:190]: I got it. I'll change over to use the common function these including the comments below on another PR soon.

File: src/EnergyPlus/FuelCellElectricGenerator.cc (hunk 3 above)
Comment (dareumnam) [reply src/EnergyPlus/FuelCellElectricGenerator.cc]: They were on B120 and B121 in csv file, and I'll update these to "Generator Produced AC Electricity Rate/Energy"

File: src/EnergyPlus/InputProcessing/InputProcessor.cc (hunk 5 above)
Comment (dareumnam) [reply src/EnergyPlus/InputProcessing/InputProcessor.cc]: Changed all "Electricity Rate" in InputProcessor.cc to uppercase.

File: src/EnergyPlus/Pumps.cc (hunk 9 above)
Comment (dareumnam) [reply src/EnergyPlus/Pumps.cc:1087]: They are in B281 & B282.

File: src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv (hunk 13 above)
Comment (dareumnam) [reply src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv]: I updated the file as your notes. Thank you for your review.

File: src/EnergyPlus/OutputReportPredefined.cc (hunk 8 above)
Comment (dareumnam) [reply src/EnergyPlus/OutputReportPredefined.cc]: Once I returned these names back to the original names.

**Comment by Myoldmopar** [comment]
CI Results Summary:
- Widespread Audit, MTD, RDD, MDD, Table: RVariable counts, no big deal
- ERR diffs appear to be in files that were *already* missing the variable name, and this just changes the string output, no problem
- EDD diffs look to be the variable names as well, and seem fine.
- JSON diffs are going to be the variable name

No meter or math diffs at all, which is key to verifying this didn't break stuff.  I am happy with these CI results.  I'll look over the changes, and then I'll run a selection of files, transitioning from 9.3 to 9.4 and making sure everything looks reasonable.

File: src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv [hunk 14]
```diff
@@ -1,114 +1,546 @@
...
-Environmental Impact Fuel Oil #1 NH3 Emissions Mass,Environmental Impact FuelOilNo1 NH3 Emissions Mass,
-Environmental Impact Fuel Oil #1 NMVOC Emissions Mass,Environmental Impact FuelOilNo1 NMVOC Emissions Mass,
-Environmental Impact Fuel Oil #1 Hg Emissions Mass,Environmental Impact FuelOilNo1 Hg Emissions Mass,
-Environmental Impact Fuel Oil #1 Pb Emissions Mass,Environmental Impact FuelOilNo1 Pb Emissions Mass,
-Environmental Impact Fuel Oil #1 Water Consumption Volume,Environmental Impact FuelOilNo1 Water Consumption Volume,
-Environmental Impact Fuel Oil #1 Nuclear High Level Waste Mass,Environmental Impact FuelOilNo1 Nuclear High Level Waste Mass,
-Environmental Impact Fuel Oil #1 Nuclear Low Level Waste Volume,Environmental Impact FuelOilNo1 Nuclear Low Level Waste Volume,
-old variable name,new variable name -- <DELETE> to delete,add variable names (before this line) and leave off units -- text in this column will add a warning
+9.3.0,9.4.0,Transition notes - some of these are EMS variable names,,,,,,
+543,543,These numbers should be the number of report variables in the following list (including deletes). Two columns/numbers.,,,,,,
```
Comment (Myoldmopar) [inline src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv:2]: I don't know why this saved with all the extra columns.  It shouldn't be a problem, but I'll still try to clean this up later.

File: src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv [hunk 15]
```diff
@@ -1,114 +1,546 @@
...
+Chiller Heater Gas Rate,Chiller Heater NaturalGas Rate,,,,,,,
+Chiller Heater Gas Energy,Chiller Heater NaturalGas Energy,,,,,,,
+Chiller Heater FuelOil#1 Rate,Chiller Heater FuelOilNo1 Rate,,,,,,,
+Chiller Heater FuelOil#1 Energy,Chiller Heater FuelOilNo1 Energy,,,,,,,
+Chiller Heater FuelOil#2 Rate,Chiller Heater FuelOilNo2 Rate,,,,,,,
+Chiller Heater FuelOil#2 Energy,Chiller Heater FuelOilNo2 Energy,,,,,,,
+Heating Coil FuelOil#1 Energy,Heating Coil FuelOilNo1 Energy,,,,,,,
+Heating Coil FuelOil#2 Rate,Heating Coil FuelOilNo2 Rate,,,,,,,
+Heating Coil FuelOil#1 Energy,Heating Coil FuelOilNo1 Energy,,,,,,,
+Heating Coil FuelOil#2 Rate,Heating Coil FuelOilNo2 Rate,,,,,,,
```
Comment (Myoldmopar) [inline src/Transition/SupportFiles/Report Variables 9-3-0 to 9-4-0.csv:545]: This. is. a. lot.

File: tst/EnergyPlus/unit/SystemReports.unit.cc [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+                        CompEnergyUse);
+   
+    EXPECT_EQ(SysHumidPropane(1), 100);
+    EXPECT_EQ(SysHCCompPropane(1), 100);
+
+    // Calculate SysTotPropane ("Air System Propane Energy")
+    ReportSystemEnergyUse();
+    EXPECT_EQ(SysTotPropane(1), 200);
+}
+} // namespace EnergyPlus
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/SystemReports.unit.cc:210]: Fixed all the other unit tests, and added a new one here. :+1:

**Review by Myoldmopar — APPROVED** [review]
This looks like a huge amount of work.  Obviously none of us are going line by line, but CI has shown that this is looking good.  I'll do some local tests, including transition tests, and we'll move forward.

**Comment by Myoldmopar** [comment]
Transitioned up files, transition diffs look as expected, and they run fine.  I think this is good to go.  If there are any other polishing things to do, we'll add them to the follow-up issue, which has already been described here.  I am ready to drop this one in.  Thanks @dareumnam and @mjwitte.

**Pull request merged**